### PR TITLE
Key Vault sample tidying

### PR DESCRIPTION
**Pull request by chlowell**
Some changes across all the samples:
- fixed typos
- fixed links
- ensured keys hello world samples delete both keys they create
- removed inaccurate numbering in print statements
- removed `if __name__ == "__main__"` from sync samples
- added license headers

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/keyvault/azure-keyvault-keys/samples/backup_restore_operations.py [hunk 1]
```diff
@@ -30,46 +27,38 @@
...
-        print("\nrun_sample has caught an error. {0}".format(e.message))
+    # Backups are good to have, if in case keys gets deleted accidentally.
+    # For long term storage, it is ideal to write the backup to a file.
+    print("\n.. Create a backup for an existing Key")
+    key_backup = client.backup_key(key.name)
+    print("Backup created for key with name '{0}'.".format(key.name))
 
-    finally:
-        print("\nrun_sample done")
+    # The rsa key is no longer in use, so you delete it.
```
Comment (iscai-msft) [inline sdk/keyvault/azure-keyvault-keys/samples/backup_restore_operations.py]: I think there should be a print statement like "\n.. Deleting Key" or something.

File: sdk/keyvault/azure-keyvault-keys/samples/hello_world_async.py [hunk 2]
```diff
@@ -53,28 +51,28 @@
...
-        print("\n3. Update a Key by name")
+        print("\n.. Update a Key by name")
         expires = datetime.datetime.utcnow() + datetime.timedelta(days=365)
         updated_ec_key = await client.update_key(ec_key.name, ec_key.version, expires=expires, enabled=False)
         print("Key with name '{0}' was updated on date '{1}'".format(updated_ec_key.name, updated_ec_key.updated))
         print("Key with name '{0}' was updated to expire on '{1}'".format(updated_ec_key.name, updated_ec_key.expires))
 
         # The RSA key is no longer used, need to delete it from the Key Vault.
-        print("\n4. Delete Key")
+        print("\n.. Delete Key")
```
Comment (iscai-msft) [inline sdk/keyvault/azure-keyvault-keys/samples/hello_world_async.py]: The sync hello_world deletes both the rsa key and the ec key, but the async version only deletes the rsa key

File: sdk/keyvault/azure-keyvault-certificates/samples/backup_restore_operations.py [hunk 3]
```diff
@@ -61,7 +61,7 @@
     # The storage account certificate is no longer in use, so you can delete it.
     client.delete_certificate(name=cert_name)
     # To ensure certificate is deleted on the server side.
-    time.sleep(30)
+    time.sleep(20)
```
Comment (iscai-msft) [inline sdk/keyvault/azure-keyvault-certificates/samples/backup_restore_operations.py]: I think there should be a print statement like "\n.. Deleting Certificate", sorry this is my bad

File: sdk/keyvault/azure-keyvault-keys/samples/list_operations.py [hunk 4]
```diff
@@ -34,76 +33,68 @@
...
-        else:
-            print("\nrun_sample has caught an error. {0}".format(e.message))
+    # Both the rsa key and ec key are not needed anymore. Let's delete those keys.
+    client.delete_key(rsa_key.name)
+    client.delete_key(ec_key.name)
 
-    finally:
-        print("\nrun_sample done")
+    # To ensure key is deleted on the server side.
+    print("\nDeleting keys...")
```
Comment (iscai-msft) [inline sdk/keyvault/azure-keyvault-keys/samples/list_operations.py]: For consistency I think it's better to be print("\n.. Deleting keys")

File: sdk/keyvault/azure-keyvault-keys/samples/recover_purge_operations.py [hunk 5]
```diff
@@ -31,61 +30,53 @@
...
-        else:
-            print("\nrun_sample has caught an error. {0}".format(e.message))
+    # Let's delete ec key now.
+    # If the keyvault is soft-delete enabled, then for permanent deletion deleted key needs to be purged.
+    client.delete_key(ec_key.name)
 
-    finally:
-        print("\nrun_sample done")
+    # To ensure key is deleted on the server side.
+    print("\nDeleting EC Key...")
```
Comment (iscai-msft) [inline sdk/keyvault/azure-keyvault-keys/samples/recover_purge_operations.py]: Should be print("\n.. Deleting EC Key") for consistency

**Pull request merged**